packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -601,6 +696,58 @@ export default{
       expect(std.err).toMatchInlineSnapshot(`""`);
     });
 
+    it("should make environment specific kv namespace for assets, even for legacy envs", async () => {
```

Comment:
I think the test above this one should have its description changed to include "even for non-legacy envs", just to be clear that this behaviour is a bit weird for non-legacy.